src/ex3/cli-todo.js

```diff
@@ -0,0 +1,117 @@
+import { Command } from "commander";
+import fetch from "node-fetch";
+import fs from "fs";
+import chalk from "chalk";
+
+const pokemonProgram = new Command();
+const path = "./todoApp.txt";
+
+async function addTodo(text) {
+  let textToAdd;
+  try {
+    const pokemon = await getPokemon(text);
+    if (pokemon && pokemon.name) {
+      textToAdd = `Catch ${pokemon.name}\n`;
+    } else {
+      textToAdd = `${text}\n`;
+    }
+    fs.appendFileSync(path, textToAdd);
+    consolSuccess(logMessage("added"));
+  } catch {
+    (err) => {
+      consoleError(logMessage("not-added"));
+      throw err;
+    };
+  }
+}
+
+async function getPokemon(text) {
+  try {
+    const response = await fetch(`https://pokeapi.co/api/v2/pokemon/${text}`);
+    const json = await response.json();
+    return json;
+  } catch {
+    (err) => {
+      throw err;
+    };
+  }
+}
+
+function getAllToDos() {
+  const todos = fs.readFileSync(path, "utf8", (err) => {
+    if (err) {
+      throw err;
+    }
+  });
+  return todos;
+}
+
+function removeTodoByIndex(index) {
+  const todos = getAllToDos();
+  const newTodoList = todos.split("\n");
+  if (newTodoList[index]) {
+    newTodoList.splice(index, 1);
+    fs.writeFileSync(path, newTodoList.join("\n"));
+    consolSuccess(logMessage("removed"));
+  } else {
+    if (isNaN(Number(index))) {
+      consoleError(logMessage("not-removed-invalid"));
+    } else {
+      consoleError(logMessage("not-removed-missing"));
+    }
+  }
+}
+
+const consoleError = (log) => {
```

Comment:
I would've extracted all console methods to an external service `console-service` which be called when needed